File: structural/facade/Facade.cs

using System;

// Subsystem A
public class SubsystemA
{
    public void OperationA()
    {
        Console.WriteLine("SubsystemA: OperationA");
    }
}

// Subsystem B
public class SubsystemB
{
    public void OperationB()
    {
        Console.WriteLine("SubsystemB: OperationB");
    }
}

// Facade class
public class Facade
{
    private SubsystemA _subsystemA;
    private SubsystemB _subsystemB;

    public Facade(SubsystemA subsystemA, SubsystemB subsystemB)
    {
        _subsystemA = subsystemA;
        _subsystemB = subsystemB;
    }

    public void Operation()
    {
        Console.WriteLine("Facade: Operation");
        _subsystemA.OperationA();
        _subsystemB.OperationB();
    }
}

// Client code
class Program
{
    static void Main()
    {
        SubsystemA subsystemA = new SubsystemA();
        SubsystemB subsystemB = new SubsystemB();
        Facade facade = new Facade(subsystemA, subsystemB);
        facade.Operation();
    }
}

File: structural/facade/facade.py

# Subsystem A
class SubsystemA:
    def operation_a(self):
        print("SubsystemA: OperationA")

# Subsystem B
class SubsystemB:
    def operation_b(self):
        print("SubsystemB: OperationB")

# Facade class
class Facade:
    def __init__(self, subsystem_a, subsystem_b):
        self._subsystem_a = subsystem_a
        self._subsystem_b = subsystem_b

    def operation(self):
        print("Facade: Operation")
        self._subsystem_a.operation_a()
        self._subsystem_b.operation_b()

if __name__ == "__main__":
    subsystem_a = SubsystemA()
    subsystem_b = SubsystemB()
    facade = Facade(subsystem_a, subsystem_b)
    facade.operation()
